include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+#ifndef _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+#define _NBL_VIDEO_S_PHYSICAL_DEVICE_LIMITS_H_INCLUDED_
+
+#include <type_traits>
+#include "nbl/asset/utils/IGLSLCompiler.h" // asset::IGLSLCompiler::E_SPIRV_VERSION
+
+
+namespace nbl::video
+{
+
+struct SPhysicalDeviceLimits
+{
+    /* Vulkan 1.0 Core  */
+    uint32_t maxImageDimension1D;
+    uint32_t maxImageDimension2D;
+    uint32_t maxImageDimension3D;
+    uint32_t maxImageDimensionCube;
+    uint32_t maxImageArrayLayers;
+    uint32_t maxBufferViewSizeTexels;
+    uint32_t maxUBOSize;
+    uint32_t maxSSBOSize;
+    //uint32_t              maxPushConstantsSize;
+    //uint32_t              maxMemoryAllocationCount;
+    //uint32_t              maxSamplerAllocationCount;
+    //VkDeviceSize          bufferImageGranularity;
+    //VkDeviceSize          sparseAddressSpaceSize;
+    //uint32_t              maxBoundDescriptorSets;
+    //uint32_t              maxPerStageDescriptorSamplers;
+    //uint32_t              maxPerStageDescriptorUniformBuffers;
+    uint32_t maxPerStageDescriptorSSBOs;
+    //uint32_t              maxPerStageDescriptorSampledImages;
+    //uint32_t              maxPerStageDescriptorStorageImages;
+    //uint32_t              maxPerStageDescriptorInputAttachments;
+    //uint32_t              maxPerStageResources;
+    //uint32_t              maxDescriptorSetSamplers;
+    uint32_t maxDescriptorSetUBOs;
+    uint32_t maxDescriptorSetDynamicOffsetUBOs;
+    uint32_t maxDescriptorSetSSBOs;
+    uint32_t maxDescriptorSetDynamicOffsetSSBOs;
+    uint32_t maxDescriptorSetImages;
+    uint32_t maxDescriptorSetStorageImages;
+    //uint32_t              maxDescriptorSetInputAttachments;
+    //uint32_t              maxVertexInputAttributes;
+    //uint32_t              maxVertexInputBindings;
+    //uint32_t              maxVertexInputAttributeOffset;
+    //uint32_t              maxVertexInputBindingStride;
+    //uint32_t              maxVertexOutputComponents;
+    //uint32_t              maxTessellationGenerationLevel;
+    //uint32_t              maxTessellationPatchSize;
+    //uint32_t              maxTessellationControlPerVertexInputComponents;
+    //uint32_t              maxTessellationControlPerVertexOutputComponents;
+    //uint32_t              maxTessellationControlPerPatchOutputComponents;
+    //uint32_t              maxTessellationControlTotalOutputComponents;
+    //uint32_t              maxTessellationEvaluationInputComponents;
+    //uint32_t              maxTessellationEvaluationOutputComponents;
+    //uint32_t              maxGeometryShaderInvocations;
+    //uint32_t              maxGeometryInputComponents;
+    //uint32_t              maxGeometryOutputComponents;
+    //uint32_t              maxGeometryOutputVertices;
+    //uint32_t              maxGeometryTotalOutputComponents;
+    //uint32_t              maxFragmentInputComponents;
+    //uint32_t              maxFragmentOutputAttachments;
```

Comment:
[GL_MAX_COLOR_ATTACHMENTS](https://opengl.gpuinfo.org/displaycapability.php?name=GL_MAX_COLOR_ATTACHMENTS)

Reply:
does the Vulkan spec count `gl_FragDepth` as a fragment output?

Reply:
hmm I think it might be [GL_MAX_DRAW_BUFFERS](https://opengl.gpuinfo.org/displaycapability.php?name=GL_MAX_DRAW_BUFFERS) instead

Reply:
>  maxColorAttachments is the maximum number of color attachments that can be used by a
subpass in a render pass. The colorAttachmentCount member of the VkSubpassDescription or
VkSubpassDescription2 structure must be less than or equal to this limit.

> maxFragmentOutputAttachments is the maximum number of output attachments which can be
written to by the fragment shader stage.

They are very closely related and according to Vulkan GPU info they are basically the same for all devices.
You could probably have a subpass with 8 color attachments but if  `maxFragmentOutputAttachments` is 4 you can only write to only 4 of those render targets in the fragment shader only.

Lemme know the exact value in GL I should query for

Reply:
it should be [GL_MAX_DRAW_BUFFERS](https://opengl.gpuinfo.org/displaycapability.php?name=GL_MAX_DRAW_BUFFERS) for `maxFragmentOutputAttachments`

and `GL_MAX_COLOR_ATTACHMENTS` for `maxColorAttachments`